### ISSUE TITLE
BenchmarkResult$aggregate() returns same performance score for all experiments

### ISSUE THREAD

**Issue by be-marc** [mlr-org/mlr3#555]
```
library(mlr3)

task = tsk("spam")
learner1 = lrn("classif.featureless")
learner2 = lrn("classif.rpart")
resampling = rsmp("subsampling", repeats = 2)

design = benchmark_grid(task, list(learner1, learner2), resampling)
bmr = benchmark(design = design, store_models = TRUE)

bmr$aggregate()

## >    nr      resample_result task_id          learner_id resampling_id iters classif.ce
## > 1:  1 <ResampleResult[21]>    spam classif.featureless            cv    10  0.2510372
## > 2:  2 <ResampleResult[21]>    spam       classif.rpart            cv    10  0.2510372

bmr$resample_result(1)$score(msr("classif.ce"))
bmr$resample_result(2)$score(msr("classif.ce"))

bmr$resample_result(1)$aggregate(msr("classif.ce"))
bmr$resample_result(2)$aggregate(msr("classif.ce"))

bmr$resample_result(1)$predictions()
bmr$resample_result(2)$predictions()
```

**Comment by mllg** [comment]
Confirmed, only devel version affected.

**Comment by mllg** [comment]
Fixed in a83cd05a7a5808a10618d0fdff7c66617e30d43b.
